enlighten/Plugins/PluginController.py

```diff
@@ -471,10 +471,19 @@ def connected_callback(self):
                 self.do_post_disconnect()
                 return
 
-            # allow the user to enable the plugin
-            self.cb_enabled.setEnabled(True)
-            self.cb_enabled.setChecked(False)
-            self.button_process.setEnabled(True)
+            module_config = self.get_current_configuration()
+            if module_config.auto_enable:
+                # this plugin has requested to auto-enable on connect
+                log.debug("auto-enabling")
+                self.cb_enabled.setEnabled(not module_config.lock_enable)
```

Comment:
Note plugins can set `self.lock_enable = True` to prevent users from deactivating the plugin once connected.